dev/ci/ci-common.sh

```diff
@@ -44,29 +44,41 @@ for overlay in "${ci_dir}"/user-overlays/*.sh; do
     . "${overlay}"
 done
 
-mathcomp_CI_DIR="${CI_BUILD_DIR}/math-comp"
-
-# git_checkout branch url dest will create a git repository
-# in <dest> (if it does not exist already) and checkout the
-# remote branch <branch> from <url>
-git_checkout()
+# [git_download project] will download [project] and unpack it
+# in [$CI_BUILD_DIR/project] if the folder does not exist already;
+# if it does, it will do nothing except print a warning (this can be
+# useful when building locally).
+# When the variable $FORCE_GIT is set to 1 or if both $FORCE_GIT and $CI
+# are undefined (local build), it uses git clone to perform the download.
+git_download()
 {
-  local _BRANCH=${1}
-  local _URL=${2}
-  local _DEST=${3}
-
-  # Allow an optional 4th argument for the commit
-  local _COMMIT=${4:-FETCH_HEAD}
-  local _DEPTH=()
-  if [ -z "${4}" ]; then _DEPTH=(--depth 1); fi
-
-  mkdir -p "${_DEST}"
-  ( cd "${_DEST}"                                                     && \
-    if [ ! -d .git ] ; then git clone "${_DEPTH[@]}" "${_URL}" . ; fi && \
-    echo "Checking out ${_DEST}"                                      && \
-    git fetch "${_URL}" "${_BRANCH}"                                  && \
-    git -c advice.detachedHead=false checkout "${_COMMIT}"            && \
-    echo "${_DEST}: $(git log -1 --format='%s | %H | %cd | %aN')" )
+  local PROJECT=$1
+  local DEST="$CI_BUILD_DIR/$PROJECT"
+  if [[ ! -n "$FORCE_GIT" && ! -n "$CI" ]]; then
+      FORCE_GIT=1
+  fi
```

Comment:
Just use `if [ -n "$FORCE_GIT" ] || [ -n '$CI" ]` in the below condition

Reply:
OK

Reply:
Wait this isn't the same, did you mean: `if [ -n "$FORCE_GIT" ] || [ ! -n "$CI" ]`?

Reply:
right
but `-z "$CI"` is cleaner than `!`

Reply:
And even this is less powerful: when `$CI` is undefined, setting `FORCE_GIT=0` would allow to download the tarball nonetheless. Whereas with this test, you would have to set `$CI` to do the same.